libtcb: Add versioning to exported symbols.

besser82 opened the pull request:
libtcb: Add versioning to exported symbols.

This change is implemented for adding some interfaces to libtcb to give it a more consumer friendly API and thus makes porting existing applications to be TCB aware easier.

* libs/libtcb.map: Declare version for exported symbols.

ldv-alt commented:
> libtcb: Add versioning to exported symbols.

Sorry, is there any particular reason to add versioning to these old symbols?

ldv-alt commented:
> > libtcb: Add versioning to exported symbols.
> 
> Sorry, is there any particular reason to add versioning to these old symbols?

Unless you're planning to add some new symbols, of cause.  If that's the case, please mention it in the commit message.

besser82 commented:
> Unless you're planning to add some new symbols, of cause. If that's the case, please mention it in the commit message.

Yes, that's point; I'm planning to add some interfaces in the near future.  Will fix the commit message on rebase.

ldv-alt commented:
> Yes, that's point; I'm planning to add some interfaces in the near future. Will fix the commit message on rebase.

You're also adding aliases.  Are you going to introduce multiple interfaces for the same name?  Otherwise, what's the purpose of adding aliases?

besser82 commented:
> > Yes, that's point; I'm planning to add some interfaces in the near future. Will fix the commit message on rebase.
> 
> You're also adding aliases. Are you going to introduce multiple interfaces for the same name? Otherwise, what's the purpose of adding aliases?

I do not have any plans to introduce mujltiple interfaces currently.  It is basically a hack, as gcc's attribute symver is expected to be placed before any function body is defined, while the traditional asm block must positioned after the definiton of the corresponding function.

For that reason I'm using the alias trickery to be able to have a uniform placement of the symver macro;  we do this the same way in libxcrypt, anyways.

ldv-alt commented:
> > Sorry, I don't quite follow. Isn't changing libtcb.map enough?
> 
> Technically, it would be, _but_ that is not enough when LTO comes into play; especially for that case the attribute symver notation has been introduced, which comes with the outlined caveats.

What's wrong with LTO?  I didn't notice any differences.  Otherwise, a lot of projects would be affected, including Linux-PAM.

besser82 commented:
> What's wrong with LTO? I didn't notice any differences. Otherwise, a lot of projects would be affected, including Linux-PAM.

I've changed it to just add the version to the version script file.  Also fixed the ChangeLog and commit to recent date.

besser82 commented:
Rebased and ready.
